### ISSUE TITLE
Floating point precision error

### ISSUE THREAD

**Issue by newcomertzc** [fangwei123456/spikingjelly#580]
There is a floating point precision error when using 0.02 at each time step. This issue can be reproduced with the following code:

```
from spikingjelly.activation_based import neuron
import torch

if_layer = neuron.IFNode(step_mode='s')

print(' -'* 6)
if_layer.reset()
print(f"init v: {if_layer.v}")
x = torch.FloatTensor([0.05])
print(f"input {x}")
print(' -'* 6)
for i in range(100):
    out = if_layer(x)
    if out == 1:
        print(f"if_layer fires at step {i}")
        
print(' -'* 6)
if_layer.reset()
print(f"init v: {if_layer.v}")
x = torch.FloatTensor([0.02])
print(f"input {x}")
print(' -'* 6)

for i in range(200):
    out = if_layer(x)
    if out == 1:
        print(f"if_layer fires at step {i}")
        
print(' -'* 6)
if_layer.reset()
print(f"init v: {if_layer.v}")
x = torch.FloatTensor([0.02])
print(f"input {x}")
print(' -'* 6)

for i in range(50):
    out = if_layer(x)
print(f"membrane at step 50:   {if_layer.v}")
print(f"membrane - 1.0:        {if_layer.v - 1.0}")
```
I suggest adding a parameter eps and setting it to 1e-6 to improve computational stability, modifying the condition to v + eps >= v_threshold. Alternatively, you could adjust the variable x used in your tutorials (https://spikingjelly.readthedocs.io/zh-cn/0.0.0.0.14/activation_based/neuron.html) to avoid any misunderstandings.

**Comment by AllenYolk** [comment]
这本质是 fp32 的精度限制——`0.02 * 100 = 2.0` 在 fp32 下并不严格等于 2.0，膜电位会被比较为 `< 1.0`，永远不发放脉冲。

几种常用方案：

1. **加大输入**。把教程里的 `x = 0.05` 改成 `x = 0.5` 或更大，让单步膜电位增量超过 fp32 的精度噪声阈值（约 1e-7）。

2. **提高 dtype**。把 `IFNode` / 输入 cast 到 `torch.float64`：

   ```python
   if_layer = neuron.IFNode(step_mode="s").double()
   x = torch.tensor([0.02], dtype=torch.float64)
   ```

3. **使用脉冲频率更高的编码**，例如 PoissonEncoder，每次发放多个脉冲，相加后误差被稀释。

教程中 `x = 0.05` 的取值是为了演示"几十步发放一次"这种稀疏场景；如果你想完整复现"100 步发放一次"这种极端例子，建议直接用方案 1 或方案 2，否则在 fp32 下会被精度噪声"吃掉"。

issue 中提到加 `eps` 的方案并非普遍需求，且会改变 fire 行为；目前不计划在代码中默认引入。

**Comment by AllenYolk** [comment]
This is fundamentally a fp32 precision limit — `0.02 * 100 = 2.0` is not bitwise equal to 2.0 in float32, so the membrane potential is compared as `< 1.0` and the neuron never fires.

Three practical workarounds:

1. **Increase the input**. Change `x = 0.05` (as in the tutorial) to `x = 0.5` or larger, so the per-step voltage increment is well above fp32's ~1e-7 precision noise.

2. **Use float64**. Cast both the IFNode and input to `torch.float64`:

   ```python
   if_layer = neuron.IFNode(step_mode="s").double()
   x = torch.tensor([0.02], dtype=torch.float64)
   ```

3. **Use higher-rate encoding** (e.g. `PoissonEncoder`). Multi-spike accumulation dilutes the error.

The tutorial uses `x = 0.05` to demonstrate a "fires every few dozen steps" sparse regime; if you really want to reproduce the "fires once in 100 steps" extreme case, go with options 1 or 2 — otherwise fp32 noise swallows the increment.

The `eps`-padding approach suggested in the original issue changes firing semantics and is not on the framework's roadmap.
